src/pages/DistributionsPage/DistributionForm.tsx

```diff
@@ -17,14 +17,34 @@ import { z } from 'zod';
 import { DISTRIBUTION_TYPE } from '../../config/constants';
```

Comment:
General feedback about changes in this file, as a first hot take: 

This is already one of the places in the code that's hardest to reason about due to the many moving parts, so ideally the Hedgey integration here would be pretty well-encapsulated rather than adding lots of conditionals amid the existing code. 

I know the ideal may not be reachable, given the way the page works, but let's discuss. e.g. the big addition starting at line 729 (on the "after" side of the diff) looks like it could potentially be moved into a separate subcomponent. Perhaps some of the new hooks can be as well.